telemetry-http-java11/src/main/java/com/newrelic/telemetry/Java11HttpPoster.java

```diff
@@ -0,0 +1,61 @@
+package com.newrelic.telemetry;
+
+import com.newrelic.telemetry.http.HttpPoster;
+import com.newrelic.telemetry.http.HttpResponse;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.nio.charset.Charset;
+import java.time.Duration;
+import java.util.Map;
+
+/** Implementation of the HttpPoster interface using an Java 11 JDK Http client. */
+public class Java11HttpPoster implements HttpPoster {
+  private final HttpClient httpClient;
+
+  /** Create a Java11HttpPoster with your own object. */
+  public Java11HttpPoster(HttpClient httpClient) {
+    this.httpClient = httpClient;
+  }
+
+  /** Create a default Java11HttpPoster with a custom call timeout. */
+  public Java11HttpPoster(Duration callTimeout) {
+    this.httpClient = HttpClient.newBuilder().connectTimeout(callTimeout).build();
+  }
+
+  /** Create a default Java11HttpPoster. */
+  public Java11HttpPoster() {
+    this.httpClient = HttpClient.newBuilder().build();
+  }
+
+  @Override
+  public HttpResponse post(URL url, Map<String, String> headers, byte[] body, String mediaType)
+      throws IOException {
+
+    try {
+      var builder =
+          HttpRequest.newBuilder(url.toURI()).POST(HttpRequest.BodyPublishers.ofByteArray(body));
+      headers.forEach((k, v) -> builder.header(k, v));
+      builder.header("Content-Type", mediaType);
+      var req = builder.build();
+
+      var response =
+          httpClient.send(
+              req, java.net.http.HttpResponse.BodyHandlers.ofString(Charset.defaultCharset()));
+
+      return of(response);
+    } catch (URISyntaxException | InterruptedException e) {
+      throw new IOException(e);
+    }
+  }
+
+  public static HttpResponse of(java.net.http.HttpResponse actual) {
```

Comment:
does this need to be public?

Reply:
I was in 2 minds about that - it makes it a trifle easier to mock / test but I'm not that fussed either way - your call.

Reply:
If we want it public, I think it needs a better name than `of`.  How about `convertToSdkResponse`?

Reply:
I'd prefer toSdkResponse() as it conserves the loose grammatical rule that prepositions should be preferred for the names of these types of converters.